packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts

```diff
@@ -160,10 +160,12 @@ export class LibraryRetrieveSourcePathExecutor extends RetrieveExecutor<
       response.data.map(lc => ({ fullName: lc.fileName, type: lc.type }))
     );
     const packageDirs = await SfdxPackageDirectories.getPackageDirectoryFullPaths();
-    for (const dir of packageDirs) {
-      filter.resolveSourceComponents(dir, { filter });
-    }
-    return filter;
+    const sourceResult = ComponentSet.fromSource({
+      fsPaths: packageDirs,
+      include: filter
+    });
+    // If no results from local source components, return the filter.
+    return sourceResult.getSourceComponents().toArray().length === 0 ? filter : sourceResult;
```

Comment:
This is important in the event that the source is not local. We would still want to make sure we have the results from the filter before continuing.

Reply:
Good catch. A nice API for this would be something like the following in the library:

```typescript
ComponentSet.fromMembers({
  members: [{ fullName: 'MyClass', type: 'ApexClass' }, { fullName: 'MyClass2', type: 'ApexClass' }],
  resolveSourcePaths: ['/path/to/force-app']
})
```

So that it always adds the members even if source wasn't found for them. CLI team requested this as well.

Reply:
Ah nice, that seems much more intuitive.